coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,376 @@
+        apply it through out the implementation of a project.
+        """
+
+
+@Formatting.subaspect
+class Length:
+    """Hold sub-aspects for file and line length.
+    """
+    class docs:
+        example = """
+        # We assume that the maximum number of characters per line is 10
+        # and that the maximum number of lines per files is 3.
+
+        def run(bear, file, filename, aspectlist):
+            return bear.run(file, filename, aspectlist)
+        """
+        example_language = 'python'
+        importance_reason = """
+        Too long lines of code and too large files result in code difficult to
+        read, understand and maintain.
+        """
+        fix_suggestions = """
+        Length issues can be fixed by writing shorter lines of code(splitting
+        long lines into multiple shorter lines); writing shorter files(
+        splitting files into modules, writing shorter methods and classes.).
+        """
+
+
+@Length.subaspect
+class LineLength:
+    """Number of character found in a line of code.
+    """
+    class docs:
+        example = """
+        print('The length of this line is 38')
+        """
+        example_langague = 'python'
+        importance_reason = """
+        Too long lines make code very difficult to read and maintain.
+        """
+        fix_suggestions = """
+        Splitting long lines of code into multiple shorter lines whenever
+        possible. Avoiding the usage of in-line language specific constructs
+        whenever they results in too long lines.
+        """
+    max_line_length = Taste[int](
+        'Maximum number of character for a line.',
+        (79, 80, 100, 120, 160), default=80)
+
+
+@Length.subaspect
+class FileLength:
+    """Number of lines found in a file.
+    """
+    class docs:
+        example = """
+        # This file would be a large file if we assume that the max number of
+        # lines per file is 10
+
+        class node:
+            def __init__(self, value, leftmostchild, leftsibling):
+                self.value=value
+                self.leftmostchild=leftmostchild
+                self.leftsibling=leftsibling
+
+        # This is example is just showing what this aspect is about, because
+        # the max number of lines per file is usually 999.
+        """
+        example_language = 'python'
+        importance_reason = """
+        Too long programs (or files) are difficult to read, maintain and
+        understand.
+        """
+        fix_suggestions = """
+        Splitting files into modules, writing shorter methods and classes.
+        """
+    max_file_length = Taste[int](
+        'Maximum number of line for a file',
+        (999,), default=999)
+
+
+@Formatting.subaspect
+class Spacing:
+    """All whitespace found between non-whitespace characters.
+    """
+    class docs:
+        example = """
+        # Here is an example of code with spacing issues including trailing
```

Comment:
it doesnt include trailing spaces any more.

also "optional/useless" should be simply "unnecessary"